### ISSUE TITLE
Update `Rocks.CodeGeneratorTest` to Map Types for Construction

### ISSUE THREAD

**Issue by JasonBock** [JasonBock/Rocks#193]
This console app is good for exploratory testing, but one issue was that types from NuGet packages would require types that satisfied generic constraints. Right now I'm just using `object` or `int`, and that doesn't always work. However, to try and figure out a way to be "generic" to somehow create types on the fly to match these constraints....I think it'll be easier to just create a type in the console app, and then, map a type that I know needs that type to the one I've defined. This will be tedious, but that'll let me test even more types.

For example, when I run the tests against CSLA, I get errors like this:

```
Error - Id: CS0311, Description: (248,44): error CS0311: The type 'object' cannot be used as type parameter 'T' in the generic type or method 'BusinessBase<T>'. There is no implicit reference conversion from 'object' to 'Csla.BusinessBase<object>'.
```

To get around this, I have to do this in `TestWithCode()`:

```csharp
static void TestWithCode()
{
	TestGenerator.Generate(new RockCreateGenerator(),
		"""
		using Rocks;
		using Csla;
		using System;

		[Serializable]
		public class Customer : BusinessBase<Customer> { }

		public static class Test
		{
			public static void Run() => Rock.Create<BusinessBase<Customer>>();
		}
		""", 
		typeof(Csla.BusinessBase<>), 
		typeof(System.ComponentModel.DataAnnotations.DataTypeAttribute));
}
```

It would be nice to do something like this:

```csharp
public class Customer : BusinessBase<Customer> { }

...

new Dictionary<Type, Type>()
{
  { typeof(Csla.BusinessBase<>), typeof(Csla.BusinessBase<Customer>) }
};
```

Basically, pass that dictionary along so my code gen can say, "when you find a type that's in this mapping dictionary, use the value in the dictionary instead."
